app/controllers/users_controller.rb

```diff
@@ -34,6 +34,25 @@ def update
     end
   end
 
+  def promote
+    users = User.where.not(staff: true)
+    @users = []
+    users.each do |user|
+      @users << "#{user.full_name} #{user.spire}"
+    end
+  end
+
+  def promote_save
+    user = User.find_by(spire: params[:user].split.last)
+    if user.nil?
+      redirect_to promote_users_path
+    elsif user.update(staff: true)
```

Comment:
can we maybe just handle the case that something goes wrong? Perhaps populate flash[:error] or whatever it is?